### PR TITLE
Removed unnecessary clfft.hpp dependency from opencl/platform.hpp

### PR DESCRIPTION
**Pull request by mlloreda**

### CODE REVIEW

File: src/backend/opencl/platform.hpp [hunk 1]
```diff
@@ -192,6 +198,6 @@ class DeviceManager
 #if defined(WITH_GRAPHICS)
         std::unique_ptr<GraphicsResourceManager> gfxManagers[MAX_DEVICES];
 #endif
-        clfftSetupData mFFTSetup;
+        clfftSetupData *mFFTSetup;
```
Comment (pavanky) [inline src/backend/opencl/platform.hpp]: Make this an `unique_ptr` so you don't have to delete it manually.

Comment (mlloreda) [inline src/backend/opencl/platform.hpp]: Good idea. Done.

File: src/backend/opencl/platform.cpp [hunk 2]
```diff
@@ -781,7 +782,8 @@ DeviceManager::~DeviceManager()
 }
 
 DeviceManager::DeviceManager()
-    : mUserDeviceOffset(0)
+    : mUserDeviceOffset(0),
+      mFFTSetup(std::unique_ptr<clfftSetupData>(new clfftSetupData))
```
Comment (umar456) [inline src/backend/opencl/platform.cpp]: this can be just 

```
mFFTSetup(new clfftSetupData)
```

Comment (mlloreda) [inline src/backend/opencl/platform.cpp]: Done.

**Comment by mlloreda** [comment]
build arrayfire linux opencl ci
build arrayfire windows ci

**Pull request merged**